### ISSUE TITLE
what's the purpose of different env files

### ISSUE THREAD

**Issue by zhcf** [apptainer/singularity#1293]
I found there are 4 files under the env, usually I just add my environment in the 99-base.sh, I want to know what's the difference of these files.

Singularity centoscaffe:/.singularity.d/env> ls
01-base.sh  90-environment.sh  95-apps.sh  99-base.sh

Thanks.

**Comment by vsoch** [comment]
They get sourced by all the different action scripts (e.g., when you run or shell) and in the order determined by their listing. For example, 90-environment.sh is the environment you define, the bases are the core that sort of work like a sandwich (I forget what is in the first, but the last one has updated for ld_library_path and path) and then 95 apps is for the Singularity implementation of  [SCIF](https://sci-f.github.io). If you had imported a Docker image, you would also see 10-docker.sh. If I remember correctly, one of those is linked to the (old standard) of having `/environment`. I'm not sure if this will change with the SIF image format, one of the sylabs maintainers can chime in.

So TLDR: a bunch of different sources for the environment getting put together into one environment.

**Comment by zhcf** [comment]
Thank you.
Whether there are some examples about SCIF and 95-apps.sh.

**Comment by vsoch** [comment]
yeah! That particular script sources environment variables relevant to apps when the container is invoked in the context of one (e.g. singularity run --app foo container.simg). If you use the SCIF client you can install the same recipes across container technologies ([see example here](https://sci-f.github.io/tutorial-really-quick-start)) with the [code here](https://github.com/sci-f/hello-world.scif).So there are many tutorials and example (with Docker as well). Here is a gentle introduction:

https://sci-f.github.io/apps/examples/tutorials/getting-started

and then some "quick start" examples

- https://sci-f.github.io/apps/examples/metrics/metrics.scif
- https://sci-f.github.io/tutorial-quick-start

and scientific examples

 - https://sci-f.github.io/apps/examples/snakemake.scif
 - https://sci-f.github.io/apps/examples/carrierseq.scif

and some very basic examples of the Singularity integration here:

- https://github.com/singularityware/singularity/tree/master/examples/apps

Either way, apps in your containers built on Singularity Hub will be discovered, for example:

https://singularity-hub.org/apps/459

And please ping me or post an issue --> https://github.com/sci-f/sci-f.github.io if you have any questions.

**Comment by zhcf** [comment]
Thanks

**Comment by dtrudg** [comment]
Hi @zhcf, and @vsoch (thanks for providing the comprehensive explanation / links). Are we good to close this, or if you feel it's important to have some further reference in the documentation, make this an issue on the docs at: https://github.com/singularityware/singularityware.github.io/issues ?

Thanks again!

**Comment by vsoch** [comment]
I think it's already discussed there, and per our discussion I think @zhcf was satisfied with the (verbosity of linkage) haha. I'll close, and @zhcf please reopen if you think necessary!
